Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/service/PersistedTasksService.java

```diff
@@ -101,46 +121,30 @@ public Tasks apply(Tasks input) {
         };
     }
 
-    private Observable<Event<Tasks>> initialiseSubject() {
-        return Observable.defer(new Func0<Observable<Event<Tasks>>>() {
-            @Override
-            public Observable<Event<Tasks>> call() {
-                if (isInitialised(taskRelay)) {
-                    return Observable.empty();
-                }
-                return localDataSource.getTasks()
-                        .flatMap(fetchFromRemoteIfOutOfDate())
-                        .switchIfEmpty(fetchFromRemote())
-                        .compose(asValidatedEvent(taskRelay.getValue()))
-                        .doOnNext(taskRelay)
-                        .publish().autoConnect();
-            }
-        });
-    }
-
-    private Func1<Tasks, Observable<Tasks>> fetchFromRemoteIfOutOfDate() {
+    private Func1<Tasks, Observable<Tasks>> fetchFromRemoteIfOutOfDate(final long actionTimeInMillis) {
         return new Func1<Tasks, Observable<Tasks>>() {
             @Override
             public Observable<Tasks> call(Tasks tasks) {
                 if (tasksDataFreshnessChecker.isFresh(tasks)) {
                     return Observable.just(tasks);
```

Comment:
That solution will not send the outdated data to the relay. The current solution either sends the fresh data or will send the outdated one while fetching from remote an then update the relay again.